index.html

```diff
@@ -1688,40 +1698,62 @@
             </dd>
           </dl>
           <ol>
-            <li>If the <a>presentation connection state</a> of
-            <code>presentationConnection</code> is not <code>connected</code>
-            or <code>connecting</code>, then abort these steps.
+            <li>Signal the intention to close the corresponding
+            <code>PresentationConnection</code> to the <a>destination browsing
+            context</a>, passing the <code>closeReason</code> to that context.
             </li>
-            <li>Set <a>presentation connection state</a> of
-            <code>presentationConnection</code> to <code>closed</code>.
+            <li>If <code>closeReason</code> is not <code>wentaway</code>, then
+            locally run the steps to <a>close the presentation connection</a>
+            with <code>presentationConnection</code>,
+            <code>closeReason</code>, and <code>closeMessage</code>.
             </li>
-            <li>
-              <a>Queue a task</a> to <a>fire</a> a <a>trusted event</a> with
-              the name <code>closed</code>, that uses the
-              <a>PresentationConnectionClosedEvent</a> interface, with the
-              <a for="PresentationConnectionClosedEvent">reason</a> attribute
-              initialized to <var>closeReason</var> and the <a for=
-              "PresentationConnectionClosedEvent">message</a> attribute
-              initialized to <var>closeMessage</var>, at
-              <code>presentationConnection</code>. The event must not bubble,
-              must not be cancelable, and has no default action.
+          </ol>
+          <p>
+            When the <a>user agent</a> is to <dfn data-lt=
+            "close the presentation connection">close a presentation
+            connection</dfn>, it MUST do the following:
+          </p>
+          <dl>
+            <dt>
+              Input
+            </dt>
+            <dd>
+              <code>presentationConnection</code>, the <a>presentation
+              connection</a> to be closed
+            </dd>
+            <dd>
+              <code>closeReason</code>, the
+              <a>PresentationConnectionClosedReason</a> describing why the
+              connection is to be closed
+            </dd>
+            <dd>
+              <code>closeMessage</code>, a human-readable message with details
+              of why the connection was closed.
+            </dd>
+          </dl>
+          <ol>
+            <li>If the <a>presentation connection state</a> of
+            <code>presentationConnection</code> is not <code>connecting</code>
+            or <code>connected</code>, then abort the remaining steps.
```

Comment:
This step may be done there to avoid queueing a task for nothing, but it should also be done once the task runs to avoid firing the "closed" event twice, for instance if the procedure is run upon receipt of a signal from the other end (which happens in the background), while the JS code is calling close() (which happens in the event loop).

I would simply move it below as first step of the task. The possibility to check the state before queueing a task seems like an implementation detail.